packages/web-pkg/src/composables/appDefaults/useAppFolderHandling.ts

```diff
@@ -29,22 +29,39 @@ export function useAppFolderHandling({
   clientService
 }: AppFolderHandlingOptions): AppFolderHandlingResult {
   const isFolderLoading = ref(false)
-  const { webdav } = clientService
+  const { webdav, graphAuthenticated } = clientService
   const { replaceInvalidFileRoute } = useFileRouteReplace()
   const { getFileInfo } = useAppFileHandling({ clientService })
   const authService = useAuthService()
   const spacesStore = useSpacesStore()
+  const sharesStore = useSharesStore()
 
   const resourcesStore = useResourcesStore()
   const { activeResources } = storeToRefs(resourcesStore)
 
   const loadFolderForFileContext = async (context: MaybeRef<FileContext>) => {
     isFolderLoading.value = true
-    resourcesStore.clearResourceList()
+
     try {
       context = unref(context)
-      const space = unref(context.space)
 
+      if (context.routeName === 'files-shares-with-me') {
+        // FIXME: this is a somewhat hacky solution to load the shared with me files.
+        // ideally we should check if there is a current folder and if not, use the
+        // folder loader to load files. unfortunately, it currently lives in the files app.
```

Comment:
As described, this is a bit of a hacky solution. My original idea was to check the presence of a current folder. If none is given, we can assume that the user is on some page where there is no actual space (Shared with me page, search results, etc). In that case we basically want the same logic as the respective folder loader for this page. However, the folder loader currently lives in the files app... Would it maybe make sense to register folder loaders via extension? @dschmidt What do you think?

Reply:
The idea of the folder loaders was definitely to make them reusable from app contexts, but I'm not sure they really fit our concept of extensions. 

At least I currently don't see a use case for third party folder loaders...

Can't we just move the folder loaders to web-pkg and use them here?

Reply:
We had a quick chat: for the upcoming release we want this quick fix in, as it's easy to backport to the stable branch.
In the long run we think it's best to move the folder loaders to web-pkg.
Making them extensions would be nice for consistency reasons and we can come up with theoretic use cases for that (i.e. when apps register new routes/views which should be reloadable ... but it seems a bit artificial at this point and we suggest to avoid the effort until it's actually needed)